### PR TITLE
gh-102190: Add additional zipfile pwd docstrings

### PR DESCRIPTION
**Pull request by dlwrnc**
- issue: https://github.com/python/cpython/issues/102190

* Issue: gh-102190

### CODE REVIEW

**Comment by arhadthedev** [comment]
@Yhg1s, @gpshead (as recently active zipfile experts)

File: Lib/zipfile/__init__.py [hunk 1]
```diff
@@ -1585,7 +1585,8 @@ def comment(self, comment):
         self._didModify = True
 
     def read(self, name, pwd=None):
-        """Return file bytes for name."""
+        """Return file bytes for name. 'pwd' is the password to decrypt
+        encrypted files."""
```
Comment (erlend-aasland) [inline Lib/zipfile/__init__.py:1589]: [The docs](https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile.read) say:

> *pwd* is the password used for encrypted files as a [bytes](https://docs.python.org/3/library/stdtypes.html#bytes) object and, if specified, overrides the default password set with [setpassword()](https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile.setpassword)

File: Lib/zipfile/__init__.py [hunk 2]
```diff
@@ -1737,7 +1738,8 @@ def extract(self, member, path=None, pwd=None):
         """Extract a member from the archive to the current working directory,
            using its full name. Its file information is extracted as accurately
            as possible. `member' may be a filename or a ZipInfo object. You can
-           specify a different directory using `path'.
+           specify a different directory using `path'.You can specify the
```
Comment (erlend-aasland) [inline Lib/zipfile/__init__.py]:
```suggestion
           specify a different directory using `path'. You can specify the
```

File: Lib/zipfile/__init__.py [hunk 3]
```diff
@@ -1750,7 +1752,8 @@ def extractall(self, path=None, members=None, pwd=None):
         """Extract all members from the archive to the current working
            directory. `path' specifies a different directory to extract to.
            `members' is optional and must be a subset of the list returned
-           by namelist().
+           by namelist().You can specify the password to decrypt all files
```
Comment (erlend-aasland) [inline Lib/zipfile/__init__.py]:
```suggestion
           by namelist(). You can specify the password to decrypt all files
```

File: Lib/zipfile/__init__.py (hunk 1 above)
Comment (gpshead) [reply Lib/zipfile/__init__.py:1589]: The docstrings in this module leave a lot to be desired overall. The existing proposed text is at least consistent with other descriptions of `pwd=` that already exist.  such as on open() "pwd is the password to decrypt files."

**Review by gpshead — APPROVED** [review]

File: Lib/zipfile/__init__.py (hunk 1 above)
Comment (erlend-aasland) [reply Lib/zipfile/__init__.py:1589]: Fine with me. I'd consider this, though:

```suggestion
        """Return file bytes for name.
        
        'pwd' is the password to decrypt encrypted files."""
```

Comment (gpshead) [reply Lib/zipfile/__init__.py:1589]: I think the automerge happened without this, feel free to make it a followup PR if desired.

**Pull request merged**